### DISCUSSION TITLE
Automatic group membership for social login

### DISCUSSION THREAD

**Discussion by Foxite** [goauthentik/authentik#12506] [Q&A] [answered]
I have friends in a discord server who I'd like to give access to certain apps. I have a working Discord login setup with an expression policy that requires users to be a member of one of a few guilds. The only problem with my current setup is that I have to manually add individual new users to a group that allows them access to the apps, which is tedious.

It would be nicer if new users who sign up using Discord are automatically added to certain groups. To do this I'm trying to use a property mapping assigned to the Discord source:

```python
return {
  "groups": ["Jellyfin User"]
}
```

But this doesn't work the way I'd like, instead of assigning a user to the existing Jellyfin User group it creates a new one and the user is added to that group. I end up with two Jellyfin User groups.

How can I add a user to a group using a property mapping, or alternatively, what's the best way to have all Discord users automatically added to a particular group?

**Comment by Foxite** [comment]
Upon further testing, the only problem with this property mapping is that a second group with an identical name gets created. It doesn't create more groups if I enroll more than 1 user with Discord. If I bind this new group to the Jellyfin application, it will Just Work.

Somehow it's distinguishing between one group called "Jellyfin User" and another group called "Jellyfin User", even though I am certain that there is truly no difference between the names, and there are no attributes about them that are different. They have distinct IDs, and that's the only difference I can see.

**Reply by rissson** [reply, marked as answer]
You need to set the group matching policy to use existing groups.

**Reply by Foxite** [reply]
This fixed it, thanks! After changing the setting on the Source object, I had to delete the second group for it to link to the real one.

**Reply by jjacobsson** [reply]
I know I am a year and change late but, is there any example or documentation or some hint on how to do this somewhere that I am just do dense to find? :D Any help appreciated.

**Reply by jjacobsson** [reply]
Oh my god I am incredibly dense. It is literally a drop down on the OAuth source! Not some scrip mapping thing! I have debugged this for hours! :D

**Comment by jeremiaspt** [comment]
> I have friends in a discord server who I'd like to give access to certain apps. I have a working Discord login setup with an expression policy that requires users to be a member of one of a few guilds. The only problem with my current setup is that I have to manually add individual new users to a group that allows them access to the apps, which is tedious.
> 
> It would be nicer if new users who sign up using Discord are automatically added to certain groups. To do this I'm trying to use a property mapping assigned to the Discord source:
> 
> ```python
> return {
>   "groups": ["Jellyfin User"]
> }
> ```
> 
> But this doesn't work the way I'd like, instead of assigning a user to the existing Jellyfin User group it creates a new one and the user is added to that group. I end up with two Jellyfin User groups.
> 
> How can I add a user to a group using a property mapping, or alternatively, what's the best way to have all Discord users automatically added to a particular group?

hello im trying to setup a login using discord, but im getting so many problems by doing the set by step documentation, do you have any way to help me on setup that?

best reggards jeremias

**Reply by Foxite** [reply]
What kind of problems are you getting?
